api/types.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+type Context struct {
+	State DebuggerState `json:"state"`
```

Comment:
Right, that's more what I had in mind. I'm certainly up for another pattern if you've seen it work before, I simply haven't had the "ah-ha" moment as you've mentioned. And GH issues aren't the best sometimes for discussion design patterns, hah.